cmd/gctsExecuteABAPUnitTests.go

```diff
@@ -0,0 +1,365 @@
+	parsingErr := parseHTTPResponseBodyXML(resp, &response)
+	if parsingErr != nil {
+		log.Entry().Warning(parsingErr)
+	}
+
+	aunitError := parseAUnitResponse(&response)
+	if aunitError != nil {
+		return aunitError
+	}
+
+	return nil
+}
+
+func parseAUnitResponse(response *runResult) error {
+	var node string
+	aunitError := false
+
+	for _, program := range response.Program {
+		log.Entry().Infof("testing class %v", program.Name)
+		for _, testClass := range program.TestClasses.TestClass {
+			log.Entry().Infof("using test class %v", testClass.Name)
+			for _, testMethod := range testClass.TestMethods.TestMethod {
+				node = testMethod.Name
+				if len(testMethod.Alerts.Alert) > 0 {
+					log.Entry().Errorf("%v - error", node)
+					aunitError = true
+				} else {
+					log.Entry().Infof("%v - ok", node)
+				}
+			}
+		}
+	}
+	if aunitError {
+		return errors.Errorf("some unit tests failed")
+	}
+	return nil
+}
+
+func getPackageList(config *gctsExecuteABAPUnitTestsOptions, telemetryData *telemetry.CustomData, client piperhttp.Sender) ([]string, error) {
+
+	type object struct {
+		Pgmid       string `json:"pgmid"`
+		Object      string `json:"object"`
+		Type        string `json:"type"`
+		Description string `json:"description"`
+	}
+
+	type objectsResponseBody struct {
+		Objects   []object      `json:"objects"`
+		Log       []gctsLogs    `json:"log"`
+		Exception gctsException `json:"exception"`
+		ErrorLogs []gctsLogs    `json:"errorLog"`
+	}
+
+	url := config.Host +
+		"/sap/bc/cts_abapvcs/repository/" + config.Repository +
+		"/getObjects?sap-client=" + config.Client
+
+	resp, httpErr := client.SendRequest("GET", url, nil, nil, nil)
+
+	defer func() {
+		if resp != nil && resp.Body != nil {
+			resp.Body.Close()
+		}
+	}()
+
+	if httpErr != nil {
+		return []string{}, errors.Wrap(httpErr, "getting repository object/package list failed")
+	} else if resp == nil {
+		return []string{}, errors.New("getting repository object/package list failed: did not retrieve a HTTP response")
+	}
+
+	var response objectsResponseBody
+	parsingErr := parseHTTPResponseBodyJSON(resp, &response)
+	if parsingErr != nil {
+		return []string{}, errors.Errorf("%v", parsingErr)
+	}
+
+	repoObjects := []string{}
+	for _, object := range response.Objects {
+		if object.Type == "DEVC" {
+			repoObjects = append(repoObjects, object.Object)
+		}
+	}
+
+	return repoObjects, nil
+}
+
+func parseHTTPResponseBodyXML(resp *http.Response, response interface{}) error {
+	if resp == nil {
+		return errors.Errorf("cannot parse HTTP response with value <nil>")
+	}
+
+	bodyText, readErr := ioutil.ReadAll(resp.Body)
+	if readErr != nil {
+		return errors.Wrap(readErr, "could not read HTTP response body")
+	}
+
+	marshalErr := xml.Unmarshal(bodyText, &response)
+	if marshalErr != nil {
+		return errors.Wrap(marshalErr, "cannot parse HTTP response as JSON")
+	}
+
+	return nil
+}
+
+func parseHTTPResponseBodyJSON(resp *http.Response, response interface{}) error {
```

Comment:
as above, is this related to the current step or would it make sense to extract into a `pkg` ?